contracts/managers/price-triggers/RSITrendingTrigger.sol

```diff
@@ -74,51 +71,53 @@ contract RSITrendingTrigger is
             "RSITrendingTrigger.constructor: Upper bound must be greater than lower bound"
         );
 
-        // Check that initial trend allocation matches one of the allocation constants
-        require(
-            _initialTrendAllocation == MAX_BASE_ASSET_ALLOCATION || _initialTrendAllocation == MIN_BASE_ASSET_ALLOCATION,
-            "RSITrendingTrigger.constructor: Initial trend allocation must match either min or max allocation values."
-        );
-
         // Set all state variables
         rsiOracleInstance = _rsiOracleInstance;
         lowerBound = _lowerBound;
         upperBound = _upperBound;
         rsiTimePeriod = _rsiTimePeriod;
-        currentTrendAllocation = _initialTrendAllocation;
+        currentTrendState = _initialTrendState;
     }
 
     /*
-     * Returns the percentage of base asset the calling Manager should allocate the RebalancingSetToken
-     * to. If RSI is above upper bound then should be 100% allocated to base asset, if
-     * RSI is below lower bound then should be 0% allocated to base asset. If in between bounds then
-     * returns the allocation of the current trend.
-     *
-     * @return             The percentage of base asset to be allocated to
+     * Since RSI does not require a confirmation leave initialTrigger function unimplemented.
+     */
+    function initialTrigger()
+        external
+    {}
+
+    /*
+     * If RSI is above upper bound then should be true, if RSI is below lower bound
+     * then should be false. If in between bounds then returns the state of the current trend.
      */
-    function retrieveBaseAssetAllocation()
+    function confirmTrigger()
         external
-        returns (uint256)
     {
         // Query RSI oracle
         uint256 rsiValue = rsiOracleInstance.read(rsiTimePeriod);
 
-        // Check RSI value is above upper bound or below lower bound
-        bool isOutsideBounds = rsiValue >= upperBound || rsiValue < lowerBound;
-
-        // If outside bounds trigger a change to currentTrendAllocation
-        if (isOutsideBounds) {
-            // If RSI greater than upper bound return max allocation of base asset
-            // Else RSI less than lower bound return min allocation of base asset
-            uint256 trendAllocation = rsiValue >= upperBound ? MAX_BASE_ASSET_ALLOCATION : MIN_BASE_ASSET_ALLOCATION;
+        // If RSI greater than upper bound return true
+        // Else if RSI less than lower bound return false
+        // Else return currentTrendState
+        bool trendState = rsiValue >= upperBound ? true : rsiValue < lowerBound ?
+            false : currentTrendState;
 
-            // Set currentTrendAllocation if trend has changed
-            if (trendAllocation != currentTrendAllocation) {
-                currentTrendAllocation = trendAllocation;
-            }
+        // Set currentTrendState if trend has changed
+        if (trendState != currentTrendState) {
```

Comment:
Let's be consistent about the if or revert pattern in the triggers. In the `MovingAverageToAssetPriceCrossover`, we revert if the state has not changed. Here we have an `if` statement. Let's pick one

I prefer to revert

Reply:
+1